elyra/pipeline/registry.py

```diff
@@ -125,19 +130,47 @@ def add_component(self, request_body):
         catalog_json = self._get_component_catalog_json()
         catalog_json['components'].append(component_json)
 
-        catalog_dir = os.path.join(os.path.dirname(__file__), "resources")
+        catalog_dir = os.path.join(os.path.dirname(__file__), self._resources_dir)
         catalog_file = os.path.join(catalog_dir, f"{self._type}_component_catalog.json")
-        with open(catalog_file, 'a') as f:
+        with open(catalog_file, 'w') as f:
             f.write(json.dumps(catalog_json))
 
-    def parse_component_details(self, component, component_name):
+    def parse_component_details(self, component, component_name=None):
         """Get component name, id, description for palette JSON"""
-        pass
+        raise NotImplementedError
 
     def parse_component_properties(self, component):
         """Get component properties for properties JSON"""
         raise NotImplementedError
 
+    def get_custom_control_id_and_type(self, parameter_type):
```

Comment:
The method name implies 'type' is going to be returned.  I would suggest just using `get_custom_control_id()`.

Reply:
Ah right, I meant to remove the `and_type`. Or, at the least, I'll have to look into it further. This is also the reason for the below comment on removing the assignment of type/format. I haven't been able to find a scenario where changing this format information really affects anything. The `custom_control_id` seems to be the key for populating the type information needed on the frontend. I'll do some research!